server/src/repositories/teammates.repository.ts

```diff
@@ -0,0 +1,112 @@
+import { EntityRepository, Repository } from 'typeorm';
+import { UserProfile } from '../entity/UserProfile';
+
+@EntityRepository(UserProfile)
+export class TeammatesRepository extends Repository<UserProfile> {
+	async getIncomingInvitations(id: string): Promise<Partial<UserProfile[]>> {
+		const user = await this
+			.createQueryBuilder('UserProfile')
+			.where('UserProfile.id = :id', { id })
+			.leftJoin('UserProfile.incomingInvites', 'user')
+			.addSelect(['user.id', 'user.firstName', 'user.lastName', 'user.avatar'])
+			.getOne();
+
+		if (!user) {
+			throw new Error('User does not exist');
+		}
+
+		return user.incomingInvites || [];
+	}
+
+	async getPendingInvitations(id: string): Promise<Partial<UserProfile[]>> {
+		const user = await this
+			.createQueryBuilder('UserProfile')
+			.where('UserProfile.id = :id', { id })
+			.leftJoin('UserProfile.pendingInvites', 'user')
+			.addSelect(['user.id', 'user.firstName', 'user.lastName', 'user.avatar'])
+			.getOne();
+
+		if (!user) {
+			throw new Error('User does not exist');
+		}
+
+		return user.pendingInvites || [];
+	}
+
+	async getTeammates(id: string): Promise<Partial<UserProfile[]>> {
+		const user = await this
+			.createQueryBuilder('UserProfile')
+			.where('UserProfile.id = :id', { id })
+			.leftJoin('UserProfile.teammates', 'user')
+			.addSelect(['user.id', 'user.firstName', 'user.lastName', 'user.avatar'])
+			.getOne();
+
+		if (!user) {
+			throw new Error('User does not exist');
+		}
+
+		return user.teammates || [];
+	}
+
+	async createInvitation(creatorId: string, teammateEmail: string): Promise<void> {
+		const creator = await this.findOne({ where: { id: creatorId } });
+		const teammate = await this.findOne({ where: { email: teammateEmail } });
+
+		if (!creator || !teammate) {
+			throw new Error(`${creator ? 'teammateEmail' : 'creatorId'} is invalid`);
+		}
+
+		const creatorPendingInvites = <UserProfile[]>await this.getPendingInvitations(creator.id);
+		const creatorTeammates = <UserProfile[]>await this.getTeammates(creator.id);
+		const creatorIncomingInvites = <UserProfile[]>await this.getIncomingInvitations(creator.id);
+		const teamMateIncomingInvites = <UserProfile[]>await this.getIncomingInvitations(teammate.id);
+
+		if (creatorTeammates.some(user => user.id === teammate.id)) {
```

Comment:
Search should be done in DB then possible. For these cases, looks like it's possible.